File: fix_reports_storage.py
import re
with open("frontend/src/lib/reportsStorage.ts", "r") as f:
    content = f.read()

content = content.replace("""      }).catch(() => {});
    }
  } catch {""", """      }).catch(() => {});
  } catch {""")

content = content.replace("""        }
      }
  } catch {
    // Ignore backend fetch error
  }""", """        }
      }
    }
  } catch {
    // Ignore backend fetch error
  }""")

with open("frontend/src/lib/reportsStorage.ts", "w") as f:
    f.write(content)

File: fix_reports_storage.js
const fs = require("fs");
let content = fs.readFileSync("frontend/src/lib/reportsStorage.ts", "utf8");

content = 'import { appendCsrfHeaders } from "./auth";\n' + content;

content = content.replace(/headers: \{\s*"Content-Type": "application\/json",\s*\}/g, 'headers: appendCsrfHeaders({ "Content-Type": "application/json" })');
content = content.replace(/method: "DELETE",\s*credentials: "include",\s*headers: \{\}/g, 'method: "DELETE",\n      credentials: "include",\n      headers: appendCsrfHeaders()');

fs.writeFileSync("frontend/src/lib/reportsStorage.ts", content);
